File: livingrimoire start here/LivinGrimoire java/src/Skills/special_skills/GamificationN.java
package Skills.special_skills;

import Auxiliary_Modules.AXGamification;
import LivinGrimoire.DiSkillV2;
import LivinGrimoire.Neuron;

public class GamificationN extends DiSkillV2 {
    private AXGamification axGamification;
    private int cost = 3;
    private DiSkillV2 skill;

    public GamificationN(DiSkillV2 skill, GamificationP rewardBank) {
        this.skill = skill;
        axGamification = rewardBank.getAxGamification();
    }

    public GamificationN setCost(int cost) {
        this.cost = cost;return this;
    }

    @Override
    public void input(String ear, String skin, String eye) {
        // engage skill only if a reward is possible
        if (axGamification.surplus(cost)){skill.input(ear, skin, eye);}
    }

    @Override
    public void output(Neuron noiron) {
        // charge reward if an algorithm is pending
        if (skill.pendingAlgorithm()) {axGamification.reward(cost);skill.output(noiron);}
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/skills/GamificationN.kt
package skills

import auxiliary_modules.AXGamification
import livinGrimoire.DiSkillV2
import livinGrimoire.Neuron

class GamificationN(skill: DiSkillV2, rewardBank: GamificationP) : DiSkillV2() {
    private val axGamification: AXGamification
    private var cost = 3
    private val skill: DiSkillV2

    init {
        this.skill = skill
        axGamification = rewardBank.getAxGamification()
    }

    fun setCost(cost: Int): GamificationN {
        this.cost = cost
        return this
    }

    override fun input(ear: String, skin: String, eye: String) {
        // engage skill only if a reward is possible
        if (axGamification.surplus(cost)) {
            skill.input(ear, skin, eye)
        }
    }

    override fun output(noiron: Neuron) {
        // charge reward if an algorithm is pending
        if (skill.pendingAlgorithm()) {
            axGamification.reward(cost)
            skill.output(noiron)
        }
    }
}